### PR TITLE
Simplify to provide single file upload.

### PR DESCRIPTION
**Pull request by chambridge**
Opened for discussion of library flow.

### CODE REVIEW

File: collector/clusterversioncollector/clusterversioncollector.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	cv, err := c.GetClusterVersion()
+	if err != nil {
+		return "", err
+	}
+	clusterID := ""
+	if cv.Spec.ClusterID != "" {
+		clusterID = string(cv.Spec.ClusterID)
+	} else {
+		err = fmt.Errorf("No cluster ID found in ClusterVersion Spec")
+	}
```
Comment (tremes) [inline collector/clusterversioncollector/clusterversioncollector.go:56]: Maybe I would simplify this if little bit. Something like: 
```
if cv.Spec.ClusterID == "" {
  return "", fmt.Errorf("No cluster ID found in ClusterVersion Spec")
} 
return string(cv.Spec.ClusterID), nil
```

File: config/config.go [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
-	Interval             time.Duration
-	Endpoint             string
-	ReportEndpoint       string
-	ReportPullingDelay   time.Duration
-	ReportMinRetryTime   time.Duration
-	ReportPullingTimeout time.Duration
-	Impersonate          string
+// Configurator An interface for handling the upload configuration
+type Configurator interface {
+	IsEnabled() bool
```
Comment (tremes) [inline config/config.go:5]: I am thinking whether we need this or not. I guess I would leave this config for the caller of the lib.  WDYT?

Comment (chambridge) [reply config/config.go:5]: Left it in as a simple way to test usage of the library to trigger the reportToLogs dry-run. But will also be useful for enable/disable when adding back the `Run()` flow.

File: collector/clusterversioncollector/clusterversioncollector.go (hunk 1 above)
Comment (chambridge) [reply collector/clusterversioncollector/clusterversioncollector.go:56]: changed. good suggestion.

File: config/config.go (hunk 2 above)
Comment (tremes) [reply config/config.go:5]: OK :-)

**Comment by tremes** [comment]
LGTM

**Review by tremes — APPROVED** [review]

**Pull request merged**